extension/chrome/elements/passphrase.ts

```diff
@@ -147,12 +147,15 @@ View.run(class PassphraseView extends View {
     const pass = String($('#passphrase').val());
     const storageType: StorageType = ($('.forget-pass-phrase-checkbox').prop('checked') || this.orgRules.forbidStoringPassPhrase()) ? 'session' : 'local';
     let atLeastOneMatched = false;
-    for (const keyinfo of this.keysWeNeedPassPhraseFor!) { // if passphrase matches more keys, it will save the pass phrase for all keys
+    const allPrivateKeys = await KeyStore.get(this.acctEmail);
+    for (const keyinfo of allPrivateKeys) { // if passphrase matches more keys, it will save the pass phrase for all keys
       const prv = await KeyUtil.parse(keyinfo.private);
       try {
         if (await KeyUtil.decrypt(prv, pass) === true) {
           await PassphraseStore.set(storageType, this.acctEmail, keyinfo.fingerprints[0], pass);
-          atLeastOneMatched = true;
+          if (this.longids.includes(keyinfo.longid)) {
```

Comment:
I'm a bit unsure myself here. I suppose, it would be more robust to check if any of the `this.longids` match any of the longids on that key (yup - there can be more than one per key - typically one for master key packet and one for subkey packet).

That looks something like this:

```ts
const keyLongids = prv.ids.map(OpenPGPKey.fingerprintToLongid)
if (this.longids.includesAnyOf(keyLongids)) {
  atLeastOneMatched = true;
}
```

But that gets even more complicated - because this will only work for OpenPGP keys but not for S/MIME keys. Plus it should be using some generic method like `KeyUtil.fingerprintToLongid` which we don't have.

@rrrooommmaaa I think this is your territory now. Could you please have a look and later finish this PR once you have time? It seems to me that the overall approach is correct but we got stuck on OpenPGP and S/MIME internals that are outside of @limonte's experties and I'm myself unsure what is the best way to proceed.

Reply:
S/MIME key only has a single longid -- "Issuer and Serial Number of a certificate" -- I'll add a test for S/MIME decryption.